src/components/ModalDrawer.tsx

```diff
@@ -79,12 +79,14 @@ const ModalDrawer = (props: ModalDrawerProps): ReactNode => {
 
 export const withModalDrawer =
   (modalDrawerProps: Partial<ModalDrawerProps> = {}) =>
-  (WrappedComponent) => {
-    const Wrapper = (props: any): ReactNode => {
-      const { isOpen, onDismiss, onExited, ...otherProps } = props;
+  <P extends { onDismiss: () => void }>(
+    WrappedComponent: ComponentType<P>
+  ): FunctionComponent<P & Pick<ModalDrawerProps, 'isOpen' | 'onExited'>> => {
+    const Wrapper = (props: P & Pick<ModalDrawerProps, 'isOpen' | 'onExited'>): ReactNode => {
+      const { isOpen, onDismiss, onExited } = props;
       return (
         <ModalDrawer isOpen={isOpen} onDismiss={onDismiss} onExited={onExited} {...modalDrawerProps}>
-          {isOpen && <WrappedComponent {...{ onDismiss, ...otherProps }} />}
+          {isOpen && <WrappedComponent {...props} />}
```

Comment:
Isn't this a change in behavior in that now all the props are being passed to `WrappedComponent`?

Reply:
Yes, it is. I had to do this to satisfy TypeScript. The alternative was casting `otherProps` to `any`. The comment in the last commit explains why I believe this change won't have any effect.